docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -379,8 +375,69 @@ def my_asset(writer: Writer):
     writer.output("hello, world!")
 
 
+defs = Definitions(assets=[my_asset], resources={"writer": WriterResource(prefix="greeting: ")})
+```
+
+### Adapting function-style I/O managers
+
+Similar to with resources, in codebases that utilize [function-style](https://docs.dagster.io/concepts/io-management/io-managers) I/O managers with the <PyObject object="@io_manager"/> decorator, you can use the <PyObject object="ConfigurableIOManagerAdapter"/> class to adapt them to the new Pythonic I/O manager pattern.
+
+The adapter subclass should list all config fields as attributes and implement a `wrapped_io_manager` property which returns the underlying I/O manager definition to adapt.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_io_adapter endbefore=end_io_adapter
+from dagster import (
+    Definitions,
+    IOManagerDefinition,
+    io_manager,
+    IOManager,
+    InputContext,
+    OutputContext,
+)
+from dagster._config.structured_config import (
+    StructuredIOManagerAdapter as ConfigurableIOManagerAdapter,
+)
+import os
+
+
+# Old code, cannot be changed for back-compat purposes
+class OldFileIOManager(IOManager):
+    def __init__(self, base_path: str):
+        self.base_path = base_path
+
+    def handle_output(self, context: OutputContext, obj):
+        with open(os.path.join(self.base_path, context.step_key, context.name), "w") as fd:
+            fd.write(obj)
+
+    def load_input(self, context: InputContext):
+        with open(
+            os.path.join(
+                self.base_path, context.upstream_output.step_key, context.upstream_output.name
+            ),
+            "r",
+        ) as fd:
+            return fd.read()
+
+
+@io_manager(config_schema={"base_path": str})
```

Comment:
If we are allow to change code but keep interface, we can develop a pattern here where we can canonicalize the config schema. We define the config schema in the Pythonic style only and programmatically generate the old school schema. I think this is the pattern we should use when we start to do integration conversion. I don't think we need to block this guide on that but you should put it on your radar.